### PR TITLE
basicfuncs: add $(names) and $(values) template functions

### PR DESCRIPTION
**Pull request by bazsi**
These template functions can be used to:
 1) convert a set of name-value pairs to a list:

      $(format-json a=list($values .SDATA.*))

 2) check if a name-value with a specific prefix exists:

      $(if '$(list-count $(names .SDATA.*)) > 0' WE-HAVE-SDATA WE-DONT-HAVE_SDATA)

Signed-off-by: Balazs Scheidler <bazsi77@gmail.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by bazsi** [comment]
@kira-syslogng test this please;

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — COMMENTED** [review]
LGTM. Please write a news entry.

**Comment by bazsi** [comment]
added a news entry.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: news/feature-3911.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+value pairs queried are specified by a value-pairs expression, just like
+with `$(format-json)`.
+
+Examples:
+
+  This expression sets the JSON array `values` to contain the list of SDATA
+  values, while the JSON array `names` would contain the associated names, in
+  the same order.
+
+  $(format-json values=list($values .SDATA.*) names=list($(names .$SDATA.*)))
```
Comment (Kokan) [inline news/feature-3911.md:12]: The example config is incorrect: `list($values .SDATA.*)`, the correct `list($(values .SDATA.*))` (missing parenthesis for value fn.
The names invocation is also not desired example (but not incorret), should be `list($(names .SDATA.*))` (the `$` is not needed).

Comment (MrAnno) [reply news/feature-3911.md]: I've fixed the example.
@Kokan Please check it.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]

**Review by Kokan — APPROVED** [review]

**Pull request merged**